test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+// result returns number of mirror services.
+func TestCheckGatewayAfterRepairEndpoints(t *testing.T) {
+	// Re-build the clientset with the source context
+	if err := TestHelper.SwitchContext(contexts["src"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["src"], err)
+	}
+	time.Sleep(time.Minute + 5*time.Second)
+	cmd := []string{"--context=" + contexts["src"], "multicluster", "check", "--wait=10s"}
+	timeout := 20 * time.Second
+	err := TestHelper.RetryFor(timeout, func() error {
+		out, err := TestHelper.LinkerdRun(cmd...)
+		if err != nil {
+			return fmt.Errorf("'linkerd multicluster check' command failed\n%s", out)
+		}
+
+		pods, err := TestHelper.KubernetesHelper.GetPods(context.Background(), "linkerd-multicluster", nil)
+		if err != nil {
+			testutil.AnnotatedFatal(t, fmt.Sprintf("failed to retrieve pods: %s", err), err)
+		}
+
+		tpl := template.Must(template.ParseFiles("testdata/check-gateway.multicluster.golden"))
+		versionErr := healthcheck.CheckProxyVersionsUpToDate(pods, version.Channels{})
+		versionErrMsg := ""
+		if versionErr != nil {
+			versionErrMsg = versionErr.Error()
+		}
+		vars := struct {
+			ProxyVersionErr string
+			HintURL         string
+			LinkName        string
+		}{
+			versionErrMsg,
+			healthcheck.HintBaseURL(TestHelper.GetVersion()),
+			"target", // name of linked cluster, not its context
+		}
+
+		var expected bytes.Buffer
+		if err := tpl.Execute(&expected, vars); err != nil {
+			testutil.AnnotatedFatal(t, fmt.Sprintf("failed to parse check.multicluster.golden template: %s", err), err)
+		}
+
+		if out != expected.String() {
+			return fmt.Errorf(
+				"Expected:\n%s\nActual:\n%s", expected.String(), out)
+		}
+		return nil
+	})
+	if err != nil {
+		testutil.AnnotatedFatal(t, fmt.Sprintf("'linkerd multicluster check' command timed-out (%s)", timeout), err)
+	}
+}
+
+// TestTargetTraffic inspects the target cluster's web-svc pod to see if the
+// source cluster's vote-bot has been able to hit it with requests. If it has
+// successfully issued requests, then we'll see log messages indicating that the
+// web-svc can't reach the voting-svc (because it's not running).
+//
+// TODO it may be clearer to invoke `linkerd diagnostics proxy-metrics` to check whether we see
+// connections from the gateway pod to the web-svc?
+func TestTargetTraffic(t *testing.T) {
+	if err := TestHelper.SwitchContext(contexts["tgt"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["tgt"], err)
+	}
+
+	ctx := context.Background()
+	// Create emojivoto in target cluster, to be deleted at the end of the test.
+	TestHelper.WithDataPlaneNamespace(ctx, "emojivoto", map[string]string{}, t, func(t *testing.T, ns string) {
```

Comment:
I'm a little confused by what's going on here. We create `emojivoto` in the `target` cluster. I expected that we'd also then need to create `emojivoto` namespace in the `source` cluster so that the service is mirrored.

However, `vote-bot.yml` and `emojivoto-no-bot.yml` both create their own `linkerd-emojivoto` namespace. So, do we need to use `WithDataPlaneNamespace` here? Or, should we remove the namespace resources from both of those manifests.

Reply:
`emojivoto-no-bot` will create its own namespace in the target cluster, that's right. We can remove the resource from the manifest, think atm that's leftover. For the target cluster, we use `WithDataPlaneNamespace()` so the namespace will be deleted at the end of the test.

In the source cluster, instead of creating the namespace programatically, we just apply `vote-bot`. It makes it easier to create the namespace without having to change the clientset context.

It would be ideal to have a nested `WithDataPlaneNamespace`, but unfortunately that'd mean two context switches: one to create the source context namespace, and one to create the target context namespace. That's why I've opted to clean source resources at the end of the test as opposed to using nested function calls.

Reply:
Yea so deleting the namespaces from the manifests makes sense which I saw you already did.